interfaces/builtin/all.go

```diff
@@ -114,6 +114,14 @@ func SanitizePlugsSlots(snapInfo *snap.Info) {
 		delete(snapInfo.Slots, slotName)
 		for _, app := range snapInfo.Apps {
 			delete(app.Slots, slotName)
+
+			var activatesOn []*snap.SlotInfo
+			for _, slot := range app.ActivatesOn {
+				if slot.Name != slotName {
+					activatesOn = append(activatesOn, slot)
+				}
+			}
+			app.ActivatesOn = activatesOn
```

Comment:
I would have expected this to be done by `snap/` package when loading from YAML.

Reply:
That would involve moving (or replicating) the `dbus` interface's `BeforePrepareSlot`, and I'm not sure it makes sense to special case dbus slot validation during the initial YAML load.

This is the place where plugs/slots that fail sanitisation are removed from the `snap.Info`, so this just extends that code to cover a new place where a slot is referenced.